diff --git a/content/base/public/nsIMessageManager.idl b/content/base/public/nsIMessageManager.idl
--- a/content/base/public/nsIMessageManager.idl                                 
+++ b/content/base/public/nsIMessageManager.idl                                 
@@ -331,18 +331,18 @@ interface nsIFrameScriptLoader : nsISupp                  
   void loadFrameScript(in AString aURL, in boolean aAllowDelayedLoad);         

   /**                                                                          
    * Removes aURL from the list of scripts which support delayed load.         
    */                                                                          
   void removeDelayedFrameScript(in AString aURL);                              
 };                                                                             

-[scriptable, builtinclass, uuid(5f552699-01a2-4f17-833b-ddb3fa0d98b2)]         
-interface nsIPermissionChecker : nsISupports                                   
+[scriptable, builtinclass, uuid(134ccbf0-5c08-11e2-bcfd-0800200c9a66)]         
+interface nsIProcessChecker : nsISupports                                      
 {                                                                              

   /**                                                                          
    * Return true iff the "remote" process has |aPermission|.  This is          
    * intended to be used by JS implementations of cross-process DOM            
    * APIs, like so                                                             
    *                                                                           
    *   recvFooRequest: function(message) {                                     
@@ -356,9 +356,32 @@ interface nsIPermissionChecker : nsISupp                   
    * then applying this permission check doesn't add any security,             
    * though it doesn't hurt anything either.                                   
    *                                                                           
    * Note: If the remote content process does *not* have |aPermission|,        
    * it will be killed as a precaution.                                        
    */                                                                          
   boolean assertPermission(in DOMString aPermission);                          

+  /**                                                                          
+   * Return true iff the "remote" process has |aManifestURL|.  This is         
+   * intended to be used by JS implementations of cross-process DOM            
+   * APIs, like so                                                             
+   *                                                                           
+   *   recvFooRequest: function(message) {                                     
+   *     if (!message.target.assertContainApp("foo")) {                        
+   *       return false;                                                       
+   *     }                                                                     
+   *     // service foo request                                                
+   *                                                                           
+   * This interface only returns meaningful data when our content is           
+   * in a separate process.  If it shares the same OS process as us,           
+   * then applying this manifest URL check doesn't add any security,           
+   * though it doesn't hurt anything either.                                                                                                           
+   *                                                                           
+   * Note: If the remote content process does *not* contain |aManifestURL|,    
+   * it will be killed as a precaution.                                        
+   */                                                                          
+  boolean assertContainApp(in DOMString aManifestURL);                         
+                                                                               
+  boolean assertAppHasPermission(in DOMString aPermission);                    
+                                                                               
 };
